query-node/mappings/src/common.ts

```diff
@@ -302,3 +305,13 @@ export function deterministicEntityId(createdInEvent: SubstrateEvent, additional
     (additionalIdentifier ? `-${additionalIdentifier}` : '')
   )
 }
+
+// Convert a BN to number without throwing an error if > Number.MAX_SAFE_INTEGER
+// add with a custom limit to maxValue if needed
+export function toNumber(value: BN, maxValue = Number.MAX_SAFE_INTEGER): number {
+  try {
+    return value.toNumber() > maxValue ? maxValue : value.toNumber()
+  } catch (e) {
```

Comment:
could BN.toNumber() throw for a different reason?

We might want to log a warning that this is happening, it could be useful for debugging since we are not storing the real values.